src/test/com/softserve/temproject/service/UserDetailsServiceImplTest.java

```diff
@@ -0,0 +1,49 @@
+package com.softserve.temproject.service;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+import com.softserve.teamproject.entity.User;
+import com.softserve.teamproject.repository.UserRepository;
+import com.softserve.teamproject.service.impl.UserDetailsServiceImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class UserDetailsServiceImplTest {
+
+  @Mock
+  UserRepository userRepository;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test(expected = UsernameNotFoundException.class)
+  public void loadUserByUsername_UserDoesNotExist_ExceptionThrown() {
+    when(userRepository.getUserByNickName(anyString())).thenReturn(null);
+    UserDetailsServiceImpl userDetailsService=new UserDetailsServiceImpl();
+    userDetailsService.setUserRepository(userRepository);
+    userDetailsService.loadUserByUsername("");
+  }
+
+  @Test
+  public void loadUserByUsername_UserExist_UserReturned() {
+    User user = new User();
+    user.setNickName("Test");
+    user.setPassword("1234");
+    when(userRepository.getUserByNickName(anyString())).thenReturn(user);
+
+    UserDetailsServiceImpl userDetailsService=new UserDetailsServiceImpl();
```

Comment:
of  ~ Role role = user.getRole();
    ~grantedAuthorities.add(new SimpleGrantedAuthority(role.getName()));